### DISCUSSION TITLE
many single Tiff to PDF

### DISCUSSION THREAD

**Discussion by loudmx** [ImageMagick/ImageMagick#6116] [Help] [answered]
### ImageMagick version

7.1

### Operating system, version and so on

Windows

### Description

I am converting multiple tiff pages to a single PDF.   in some cases I have 50+ pages and I want then in a certain order. Aside from typing  a long command line :    magick  1.tif  2.tif 3.tif ..... output.pdf  
is my only option to rename the files with page # sequence at begining  001a.tif, 002c.tif 003b.tif etc.. 
 and use the       magick *.tif output.pdf  

It would be cool if I could supply a file file contaiing a list for input

i.e.  magick -filelist list.txt output.pdf      where list.txt contains the items to combine into the PDF.

Maybe something to add to future version?

**Comment by fmw42** [comment]
I tried

`magick *.tif[3,5,1,2...as desired] output.pdf`

but in my tests that did not work. It ended up in the alphanumeric order as listed in my directory.


Another thought would be to make a List of Clones of the images you want

List_of_Clones="3,5,1,2...as desired"

Then find out how many images in the tiff, say N and let M=N-1

`magick *.tif ( -clone List_of_Clones ) -delete 0-M output.pdf`



Example in Unix syntax:

```
clones="4,2,3,0,1"
magick *.jpg \( -clone $clones \) -delete 0-4 x.pdf

```

That seems to work.

But you will need to have enough RAM to hold all your images times 2.

**Comment by snibgo** [comment, marked as answer]
> It would be cool if I could supply a file file contaiing a list for input

You can, like this:
```
magick @filelist.txt out.tiff
```
where the file `filelist.txt` might contain:
```
toes.png
toes_holed.png
```
... and so on.

**Reply by loudmx** [reply]
Yes, that works. I needed to enclose my filename in "" due to spaces, but it does the trick.   I have to convert a few 100k pages down to about 100k documents.   

Thank for the help!

**Comment by Jmuccigr** [comment]
I'd suggest using a different tool, [img2pdf](https://github.com/josch/img2pdf). It's designed to put images into PDF files without modifying the images. Super fast and preserves your images (usually).
